### ISSUE TITLE
Device internet block

### ISSUE THREAD

**Issue by jacquesvzijp** [Vaskivskyi/ha-asusrouter#805]
### The problem

When using the device internet block I consistently get error message that str variable is expected. See log.

### Your device model

Rt-ac88U

### Firmware type

Merlin

### Firmware version

3.0.0.4.386.7

### Integration version with the issue

29

### Method of the integration installation

HACS

### What version and type of Home Assistant installation do you use

2024.02

### Is there anything useful in the logs?

```txt
logger:%20homeassistant.components.automation.test_jack_network_block%0ASource%3A%20components/automation/__init__.py:666%0AIntegration:%20Automatisering%20(documentation,%20issues)%0AFirst%20occurred:%2011:29:00%20(29%20occurrences)%0ALast%20logged:%2011:57:00%0A%0AWhile%20executing%20automation%20automation.test_jack_network_block%0ATraceback%20(most%20recent%20call%20last):%0A%20%20File%20%22/usr/src/homeassistant/homeassistant/components/automation/__init__.py%22,%20line%20666,%20in%20async_trigger%0A%20%20%20%20return%20await%20self.action_script.async_run(%0A%20%20%20%20%20%20%20%20%20%20%20%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%0A%20%20File%20%22/usr/src/homeassistant/homeassistant/helpers/script.py%22,%20line%201600,%20in%20async_run%0A%20%20%20%20return%20await%20asyncio.shield(run.async_run())%0A%20%20%20%20%20%20%20%20%20%20%20%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%0A%20%20File%20%22/usr/src/homeassistant/homeassistant/helpers/script.py%22,%20line%20435,%20in%20async_run%0A%20%20%20%20await%20self._async_step(log_exceptions=False)%0A%20%20File%20%22/usr/src/homeassistant/homeassistant/helpers/script.py%22,%20line%20487,%20in%20_async_step%0A%20%20%20%20self._handle_exception(%0A%20%20File%20%22/usr/src/homeassistant/homeassistant/helpers/script.py%22,%20line%20512,%20in%20_handle_exception%0A%20%20%20%20raise%20exception%0A%20%20File%20%22/usr/src/homeassistant/homeassistant/helpers/script.py%22,%20line%20485,%20in%20_async_step%0A%20%20%20%20await%20getattr(self,%20handler)()%0A%20%20File%20%22/usr/src/homeassistant/homeassistant/helpers/script.py%22,%20line%20723,%20in%20_async_call_service_step%0A%20%20%20%20response_data%20=%20await%20self._async_run_long_action(%0A%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%0A%20%20File%20%22/usr/src/homeassistant/homeassistant/helpers/script.py%22,%20line%20685,%20in%20_async_run_long_action%0A%20%20%20%20return%20long_task.result()%0A%20%20%20%20%20%20%20%20%20%20%20%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%0A%20%20File%20%22/usr/src/homeassistant/homeassistant/core.py%22,%20line%202279,%20in%20async_call%0A%20%20%20%20response_data%20=%20await%20coro%0A%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%0A%20%20File%20%22/usr/src/homeassistant/homeassistant/core.py%22,%20line%202316,%20in%20_execute_service%0A%20%20%20%20return%20await%20target(service_call)%0A%20%20%20%20%20%20%20%20%20%20%20%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%0A%20%20File%20%22/config/custom_components/asusrouter/router.py%22,%20line%20759,%20in%20async_service_device_internet_access%0A%20%20%20%20await%20self.bridge.async_pc_rule(raw=service.data)%0A%20%20File%20%22/config/custom_components/asusrouter/bridge.py%22,%20line%20649,%20in%20async_pc_rule%0A%20%20%20%20result%20=%20await%20self.api.async_set_state(rule)%0A%20%20%20%20%20%20%20%20%20%20%20%20%20%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%0A%20%20File%20%22/usr/local/lib/python3.12/site-packages/asusrouter/asusrouter.py%22,%20line%20765,%20in%20async_set_state%0A%20%20%20%20result%20=%20await%20set_state(%0A%20%20%20%20%20%20%20%20%20%20%20%20%20%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%0A%20%20File%20%22/usr/local/lib/python3.12/site-packages/asusrouter/modules/state.py%22,%20line%20155,%20in%20set_state%0A%20%20%20%20return%20await%20submodule.set_state(%0A%20%20%20%20%20%20%20%20%20%20%20%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%0A%20%20File%20%22/usr/local/lib/python3.12/site-packages/asusrouter/modules/parental_control.py%22,%20line%2084,%20in%20set_state%0A%20%20%20%20return%20await%20set_rule(callback,%20state,%20**kwargs)%0A%20%20%20%20%20%20%20%20%20%20%20%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%0A%20%20File%20%22/usr/local/lib/python3.12/site-packages/asusrouter/modules/parental_control.py%22,%20line%20143,%20in%20set_rule%0A%20%20%20%20service_arguments%20=%20write_pc_rules(current_rules)%0A%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%5E%0A%20%20File%20%22/usr/local/lib/python3.12/site-packages/asusrouter/modules/parental_control.py%22,%20line%20285,%20in%20write_pc_rules%0A%20%20%20%20data%5Bkey%5D%20=%20%22%3E%22.join(%0A%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%5E%5E%5E%5E%5E%5E%5E%5E%5E%0ATypeError:%20sequence%20item%200:%20expected%20str%20instance,%20NoneType%20found%0A
```

**Comment by Vaskivskyi** [comment]
Please, check whether the issue is resolved in the latest `0.30.0` release. If not, reopen this issue

**Comment by jacquesvzijp** [comment]
Thanks a lot, I confirm it is working!
Jacques
